### PR TITLE
Project Management System Create API

### PR DESCRIPTION
**Pull request by chetankolte1**
- Added Project Management System Create API 
- Added Flyway script 
- Added create api validations, enrichment
- Added MDMS Util file for MDMS validations
- Added Producer
- Update web models
- Updated contract

### CODE REVIEW

File: backend/project-management-system/src/main/java/org/egov/works/enrichment/ProjectEnrichment.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+            for (Document document: project.getDocuments()) {
+                document.setId(UUID.randomUUID().toString());
+                document.setAuditDetails(auditDetails);
+            }
+
+            project.getAddress().setAuditDetails(auditDetails);
+            project.getAddress().setId(UUID.randomUUID().toString());
+
+            project.getAddress().getLocality().setId(UUID.randomUUID().toString());
```
Comment (subhashini-egov) [inline backend/project-management-system/src/main/java/org/egov/works/enrichment/ProjectEnrichment.java:42]: @chetankolte1 what if this locality already exists in the DB?

File: backend/project-management-system/src/main/java/org/egov/works/enrichment/ProjectEnrichment.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+            List<Boundary> children = project.getAddress().getLocality().getChildren();
```
Comment (subhashini-egov) [inline backend/project-management-system/src/main/java/org/egov/works/enrichment/ProjectEnrichment.java:44]: Not following this.

**Review by subhashini-egov — COMMENTED** [review]
@chetankolte1 please review this once with @vguptatw once. I have made my comments.

File: backend/project-management-system/src/main/resources/db/migration/main/V20221219112900__create_pms_tables.sql [hunk 3]
```diff
@@ -0,0 +1,100 @@
+CREATE TABLE eg_pms_project(
+id                           character varying(64),
+tenantid                     character varying(64) NOT NULL,
+projecttype                  character varying(64),
```
[the remark this answers is not in the document]
Comment (chetankolte1) [reply backend/project-management-system/src/main/resources/db/migration/main/V20221219112900__create_pms_tables.sql:4]: Hi,

All the column types are renamed in readable form. I did the corresponding changes in PR-#228 (https://github.com/egovernments/DIGIT-Works/pull/228). All the latest code is in PR-#228

File: backend/project-management-system/src/main/resources/db/migration/main/V20221219112900__create_pms_tables.sql [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+subprojecttype               character varying(64),
```
[the remark this answers is not in the document]
Comment (chetankolte1) [reply backend/project-management-system/src/main/resources/db/migration/main/V20221219112900__create_pms_tables.sql]: Hi,

All the column types are renamed in readable form. I did the corresponding changes in PR-#228 (https://github.com/egovernments/DIGIT-Works/pull/228). All the latest code is in PR-#228

File: backend/project-management-system/src/main/resources/db/migration/main/V20221219112900__create_pms_tables.sql [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+department                   character varying(64),
+description                  character varying(256),
+reference_id                 character varying(100),
+startdate                    bigint NOT NULL,
```
[the remark this answers is not in the document]
Comment (chetankolte1) [reply backend/project-management-system/src/main/resources/db/migration/main/V20221219112900__create_pms_tables.sql:9]: Hi,

Did the corresponding changes in PR-#228 (https://github.com/egovernments/DIGIT-Works/pull/228). All the latest code is in PR-#228

**Comment by subhashini-egov** [comment]
@chetankolte1 told me to look at 228. Closing this.

**Pull request closed without merging**